es6-shim.js

```diff
@@ -720,6 +728,13 @@
     defineProperty(Array, 'from', ArrayShims.from, true);
   }
 
+  // Given an argument x, it will return an IteratorResult object,
+  // with value set to x and done to false.
+  // Given no arguments, it will return an iterator completion object.
+  var iterator_result = function (x) {
+    return { value: x, done: !arguments.length };
```

Comment:
Please use `arguments.length === 0` so as to not rely on number-boolean coercion.